src/core.js

```diff
@@ -101,6 +101,14 @@ var Geocoder = L.Control.extend({
         'As of v1.6.0, the `latlng` option is deprecated. It has been renamed to `focus`. `latlng` will be removed in a future version.');
     }
 
+    // Deprecation warnings for Mapzen hosted service.
+    // Make sure people aware of Mapzen hosted services are going down.
+    var mapzenHostedServiceUrl = 'https://search.mapzen.com';
+
+    if (options.url.indexOf(mapzenHostedServiceUrl)) {
```

Comment:
This will fail if `options` or `options.url` is undefined.

The other warnings will check to make sure `options` is defined (e.g. someone has actually specified options). It should also throw the warning if `options.url` isn't set, because it would then default to Mapzen Search.

Reply:
do you mind if I do check against `this.options` directly  after default options were merged with users?